feat: add API mock tests

muslldev opened the pull request:
- Mock username
- Mock token

muslldev committed:
fix: remove the possibility of substituting garbage in key "accessToken"

muslldev committed:
feat: add API mock tests

- Mock username
- Mock token

vIad17 commented on src/layouts/InstrumentLayout/components/Header/Header.tsx:
```diff
@@ -80,9 +78,41 @@ const Header = ({ className = '' }: HeaderProps) => {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ username: 'Artem', password: '1234' })
+    });
```
давай вместо фетчей сразу использовать сторонние либы. Видел ты axios подключил в другом mr-е. Здесь тоже можно его подключить, сразу сделать по красоте

vIad17 commented on src/layouts/InstrumentLayout/components/Header/Header.tsx:
```diff
@@ -80,9 +78,41 @@ const Header = ({ className = '' }: HeaderProps) => {
 
   const { input, inputs, selectInput, selectedInputId } = useMIDIInputs();
 
+  async function getUserInfo() {
+    const res = await fetch('/users/0');
+    const data = await res.json();
+    localStorage.setItem('username', data.username);
+  }
+
+  async function login() {
+    const res = await fetch('/login', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ username: 'Artem', password: '1234' })
+    });
+
+    if (!res.ok) {
+      throw new Error('Login failed');
+    }
+
+    const data = await res.json();
+    localStorage.setItem('accessToken', data.accessToken);
+  }
+
   useEffect(() => {
-    const midiInput = localStorage.getItem('midi-input');
-    midiInput && selectInput(midiInput);
+    async function init() {
+      const midiInput = localStorage.getItem('midi-input');
+      midiInput && selectInput(midiInput);
+
+      const token = localStorage.getItem('accessToken');
+      if (!token) {
+        await login();
+      }
+
+      await getUserInfo();
+    }
+
+    init();
```
так, этот useEffect трогать не надо, он отвечает за выбор midi устройства для инпута. Всю логику с токенами нужно вынести в отдельный useEffect. Иначе мы каждый раз будет запрашивать пользователя при выборе нового midi устройства

vIad17 commented on src/shared/icons/Icon.tsx:
лезешь куда ни попадя со своим автоформаттером))
Так-то классно, что ты делаешь то, на что я время от времени забиваю (форматирование), за это спасибо. Но учти, что могут быть конфликты из-за этого, т.к. над секвенсором со стороны внутренней логики также ведётся работа. Так что постарайся сильно не распоясываться)

vIad17 commented on .env.mock:
добавь ещё обычнй `.env`

vIad17 replied:
не, я имел ввиду добавить его на проект, а не в гитигнор. Там и не будет никаких данных для прода, так что не вижу проблем хранить его

vIad17 commented on public/mockServiceWorker.js:
```diff
@@ -0,0 +1,349 @@
+/* eslint-disable */
+/* tslint:disable */
```
ну не, добавь сразу ts). Тут нужна помощь, зови, помогу с переводом всего кода на ts. А eslint, насколько я помню, и так временно отключён

vIad17 replied:
не убрано

vIad17 commented on yarn.lock:
бляха. Сможешь закинуть yarn.lock в гитигнор? не уследил я за ним

vIad17 commented on package.json:
```diff
@@ -3,7 +3,8 @@
   "scripts": {
-    "start": "vite",
+    "start": "vite dev",
+    "start:mock": "vite dev --mode=mock",
```
сможешь start:mock вынести также в README, чтобы другие тоже знали?

vIad17 reviewed (commented):
моки сделаны норм, библиотечка неплохая. Есть замечания. Их исправить и збс

muslldev committed:
fix: move getUserInfo in separate useEffect

muslldev committed:
Merge branch 'hotfix/refresh-token' into feature/mock-tests

muslldev committed:
fix: add PO's wishes 🔧

- Add yarn.lock in .gitignore
- Add Mock Tests instruction in README.md
- Add opportunity of compilation TS worker to JS worker

vIad17 commented on src/shared/lib/msw/mockServiceWorker.d.ts:
```diff
@@ -0,0 +1,57 @@
+
+declare function sendToClient(
+  client: Client,
+  message: any,
```
избавься от any

muslldev committed:
fix: add .env with dev settings and remove comments in TS MSW config
